packages/datadog-plugin-next/src/index.js

```diff
@@ -0,0 +1,132 @@
+'use strict'
+
+function createWrapHandleRequest (tracer, config) {
+  return function wrapHandleRequest (handleRequest) {
+    return function handleRequestWithTrace (req, res, pathname, query) {
+      return trace(tracer, config, req, res, () => handleRequest.apply(this, arguments))
+    }
+  }
+}
+
+function createWrapHandleApiRequest (tracer, config) {
+  return function wrapHandleApiRequest (handleApiRequest) {
+    return function handleApiRequestWithTrace (req, res, pathname, query) {
+      return trace(tracer, config, req, res, () => {
+        const promise = handleApiRequest.apply(this, arguments)
+
+        promise.then(handled => {
+          if (!handled) return
+
+          const page = getPageFromPath(pathname, this.dynamicRoutes)
+
+          addPage(req, page)
+        })
+
+        return promise
+      })
+    }
+  }
+}
+
+function createWrapRenderToHTML (tracer, config) {
+  return function wrapRenderToHTML (renderToHTML) {
+    return function renderToHTMLWithTrace (req, res, pathname, query, parsedUrl) {
+      return trace(tracer, config, req, res, () => renderToHTML.apply(this, arguments))
+    }
+  }
+}
+
+function createWrapRenderToHTMLWithComponents (tracer, config) {
+  return function wrapHandleRenderToHTMLWithComponents (renderToHTMLWithComponents) {
+    return function renderToHTMLWithComponentsWithTrace (req, res, page) {
+      addPage(req, page)
+
+      return renderToHTMLWithComponents.apply(this, arguments)
+    }
+  }
+}
+
+function getPageFromPath (page, dynamicRoutes = []) {
+  for (const dynamicRoute of dynamicRoutes) {
+    if (dynamicRoute.page.startsWith('/api') && dynamicRoute.match(page)) {
+      return dynamicRoute.page
+    }
+  }
+
+  return page
+}
+
+function trace (tracer, config, req, res, handler) {
+  const scope = tracer.scope()
+
+  if (req._datadog_next) return scope.activate(req._datadog_next.span, handler)
+
+  const childOf = scope.active()
```

Comment:
Doesn't `startSpan` do this implicitly?

Reply:
It doesn't, only `tracer.trace` does it implicitly.